lib/rules/prefer-implicit-assert.ts

```diff
@@ -0,0 +1,177 @@
+		node.parent?.parent?.parent.property.type === AST_NODE_TYPES.Identifier &&
+		PRESENCE_MATCHERS.includes(node.parent.parent.parent.property.name)
+	);
+};
+
+const usesNotPresenceAssertion = (
+	node: TSESTree.Identifier | TSESTree.Node,
+	isAsyncQuery: boolean
+) => {
+	if (isAsyncQuery) {
+		return (
+			isMemberExpression(node.parent?.parent?.parent?.parent) &&
+			node.parent?.parent?.parent?.parent.property.type ===
+				AST_NODE_TYPES.Identifier &&
+			node.parent.parent.parent.parent.property.name === 'not' &&
+			isMemberExpression(node.parent.parent.parent.parent.parent) &&
+			node.parent.parent.parent.parent.parent.property.type ===
+				AST_NODE_TYPES.Identifier &&
+			ABSENCE_MATCHERS.includes(
+				node.parent.parent.parent.parent.parent.property.name
+			)
+		);
+	}
+	return (
+		isMemberExpression(node.parent?.parent?.parent) &&
+		node.parent?.parent?.parent.property.type === AST_NODE_TYPES.Identifier &&
+		node.parent.parent.parent.property.name === 'not' &&
+		isMemberExpression(node.parent.parent.parent.parent) &&
+		node.parent.parent.parent.parent.property.type ===
+			AST_NODE_TYPES.Identifier &&
+		ABSENCE_MATCHERS.includes(node.parent.parent.parent.parent.property.name)
+	);
+};
+
+const reportError = (
+	context: Readonly<
+		TSESLint.RuleContext<'preferImplicitAssert', []> & {
+			settings: TestingLibrarySettings;
+		}
+	>,
+	node: TSESTree.Identifier | TSESTree.Node | undefined,
+	queryType: string
+) => {
+	if (node) {
+		return context.report({
+			node,
+			messageId: 'preferImplicitAssert',
+			data: {
+				queryType,
+			},
+		});
+	}
+};
+
+export default createTestingLibraryRule<Options, MessageIds>({
+	name: RULE_NAME,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			description:
+				'Suggest using implicit assertions for getBy* & findBy* queries',
+			recommendedConfig: {
+				dom: false,
+				angular: false,
+				react: false,
+				vue: false,
+				marko: false,
+			},
+		},
+		messages: {
+			preferImplicitAssert:
+				"Don't wrap `{{queryType}}` query with `expect` & presence matchers like `toBeInTheDocument` or `not.toBeNull` as `{{queryType}}` queries fail implicitly when element is not found",
+		},
+		schema: [],
+	},
+	defaultOptions: [],
+	create(context, _, helpers) {
+		const findQueryCalls: TSESTree.Identifier[] = [];
+		const getQueryCalls: TSESTree.Identifier[] = [];
+
+		return {
+			'CallExpression Identifier'(node: TSESTree.Identifier) {
+				if (helpers.isFindQueryVariant(node)) {
+					findQueryCalls.push(node);
+				}
+				if (helpers.isGetQueryVariant(node)) {
+					getQueryCalls.push(node);
+				}
+			},
+			'Program:exit'() {
+				let isAsyncQuery = true;
+				findQueryCalls.forEach((queryCall) => {
+					const node: TSESTree.Identifier | TSESTree.Node | undefined =
+						isCalledUsingSomeObject(queryCall) ? queryCall.parent : queryCall;
+
+					if (node) {
+						if (isCalledInExpect(node, isAsyncQuery)) {
+							if (usesPresenceAssertion(node, isAsyncQuery))
```

Comment:
The `helpers` arg already provides a `isPresenceAssert` util. Can you use that one instead?

Reply:
done